### PR TITLE
IPv6 support

### PR DESCRIPTION
**Pull request by CyberDem0n**
fixes https://github.com/zalando/patroni/issues/1121

### CODE REVIEW

File: patroni/api.py [hunk 1]
```diff
@@ -532,8 +533,36 @@ def __get_ssl_options(config):
...
+                if sock:
+                    sock.close()
+        return False
+
+    def __httpserver_init(self, host, port):
+        dual_stack = self.__has_dual_stack()
+        if host == '':
+            host = None
+        if host is None and dual_stack:
+            host = '::'
```
Comment (erthalion) [inline patroni/api.py]: Is this required? Python docs says, that if

```
By passing None as the value of host and port, you can
pass NULL to the underlying C API.
```

and in this case:

```
the network address will be set to the loopback interface address
(INADDR_LOOPBACK for IPv4 addresses, IN6ADDR_LOOPBACK_INIT
for IPv6 address)
```

So looks like with `None` result would be the same.

Comment (CyberDem0n) [reply patroni/api.py]: You are right. Will remove it.

File: patroni/utils.py [hunk 2]
```diff
@@ -342,3 +343,17 @@ def split_host_port(value, default_port):
     t = value.rsplit(':', 1)
     t.append(default_port)
     return t[0], int(t[1])
+
+
+def uri(proto, netloc, path='', user=None):
+    host, port = netloc if isinstance(netloc, (list, tuple)) else split_host_port(netloc, 0)
+    if host and host[0] != '[' and host[-1] != ']':
+        try:
+            socket.inet_pton(socket.AF_INET6, host)
```
Comment (erthalion) [inline patroni/utils.py]: Looks like Windows support for `inet_pton` was added only in 3.4, which makes supported versions a bit different.

**Comment by erthalion** [comment]
:+1:

**Comment by erthalion** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**